src/Workleap.Extensions.Mongo/MongoServiceCollectionExtensions.cs

```diff
@@ -37,6 +43,86 @@ public static MongoBuilder AddMongo(this IServiceCollection services, Action<Mon
         return new MongoBuilder(services);
     }
 
+    /// <summary>
+    /// Adds Configuration based support for Mongo Collection Names, Index Providers and Class maps.
+    /// Each Collection requires a name. Index Providers are optional, as are Class maps.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="assemblies">Assemblies in which we can find the implementations of IMongoCollectionConfiguration</param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static MongoBuilder AddCollectionConfigurations(this MongoBuilder builder, params Assembly[] assemblies)
+    {
+        var configurationTypes = GetMongoCollectionConfigurationTypes(assemblies);
+
+        foreach (var configurationType in configurationTypes)
+        {
+            var (concreteType, configurationInterface) = configurationType;
+
+            if (RegisteredConfigurations.Contains(concreteType))
+            {
+                continue;
```

Comment:
Should we throw if there is multiple registration for the same type? It's probably a bug the consummer should be aware,.